### PR TITLE
Add check that server and hz-cli versions match [HZ-1173]

### PR DESCRIPTION
**Pull request by frant-hartm**
Jet compatibility guarantees require the hz-cli and server versions to
be the same minor version. This commit adds check for all hz-cli
operations.

It might be possible that we need to ignore the check (e.g. for testing
or debugging), for such case a `--ignore-version-mismatch` option is
provided.

Fixes https://github.com/hazelcast/hazelcast/issues/20019

Breaking changes (list specific methods/types/messages):
* previously it was possible to submit a job using a different version of `hz-cli` from the server version. Generally, it would work when there were no change in serialized form of the pipeline. There is no guarantee. Now we fail early.

Note to the reviewers: It might be easier to look at the commits separately as the 2nd commit 7551bd6f87d7a790d07eba69fc47845ba005e4ea only refactors the `HazelcastCommandLine` class.

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Comment by frant-hartm** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/client/console/HazelcastCommandLine.java [hunk 1]
```diff
@@ -700,6 +698,11 @@ public Targets convert(String value) {
         }
     }
 
+    public static class TargetsMixin {
+
+
+    }
+
```
Comment (mdumandag) [inline hazelcast/src/main/java/com/hazelcast/client/console/HazelcastCommandLine.java:705]: Is this a leftover?

File: hazelcast/src/main/java/com/hazelcast/client/console/HazelcastCommandLine.java [hunk 2]
```diff
@@ -462,16 +465,30 @@ private void runWithHazelcast(TargetsMixin targets, Verbosity verbosity, boolean
...
             consumer.accept(hz);
         } finally {
             hz.shutdown();
         }
     }
 
-    private HazelcastInstance getHazelcastClient(boolean retryClusterConnectForever) {
-        return uncheckCall(() -> hzClientFn.apply(getClientConfig(retryClusterConnectForever)));
+    private HazelcastInstance getHazelcastClient(boolean retryClusterConnectForever, boolean ignoreVersionMismatch) {
+        return uncheckCall(() -> {
```
[the remark this answers is not in the document]
Comment (frant-hartm) [reply hazelcast/src/main/java/com/hazelcast/client/console/HazelcastCommandLine.java:456]: cbf823721fb

File: hazelcast/src/main/java/com/hazelcast/client/console/HazelcastCommandLine.java (hunk 1 above)
Comment (frant-hartm) [reply hazelcast/src/main/java/com/hazelcast/client/console/HazelcastCommandLine.java]: 99128179519

**Review by mdumandag — APPROVED** [review]

**Review by TomaszGaweda — APPROVED** [review]

**Comment by frant-hartm** [comment]
BasicNestedFieldsTest failure is unrelated.

**Pull request merged**